agent/src/util/registration.rs

```diff
@@ -0,0 +1,460 @@
+use super::constants::{
+    CLOSE_DISCOVERY_HANDLER_CONNECTION_CHANNEL_CAPACITY, ENABLE_DEBUG_ECHO_LABEL,
+};
+use akri_discovery_utils::discovery::v0::{
+    register_discovery_handler_request::EndpointType,
+    registration_server::{Registration, RegistrationServer},
+    Empty, RegisterDiscoveryHandlerRequest,
+};
+use akri_shared::{
+    os::env_var::{ActualEnvVarQuery, EnvVarQuery},
+    uds::unix_stream,
+};
+use futures::TryStreamExt;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Instant;
+use tokio::sync::broadcast;
+use tonic::{transport::Server, Request, Response, Status};
+
+/// Map of `DiscoveryHandlers` of the same type (registered with the same name) where key is the endpoint of the
+/// Discovery Handler and value is `DiscoveryDetails`.
+pub type SubsetDiscoveryHandlerMap = HashMap<DiscoveryHandlerEndpoint, DiscoveryDetails>;
+
+/// Map of all registered `DiscoveryHandlers` where key is `DiscoveryHandler` name and value is a map of all
+/// `DiscoveryHandlers` with that name.
+pub type RegisteredDiscoveryHandlerMap = Arc<Mutex<HashMap<String, SubsetDiscoveryHandlerMap>>>;
```

Comment:
nit - use DiscoveryHandlerName instead of String

Reply:
fixed in 
1ff5a0f